command/agent/config_parse_test.go

```diff
@@ -620,6 +575,9 @@ func (c *Config) addDefaults() {
 	if c.Consul == nil {
 		c.Consul = config.DefaultConsulConfig()
 		c.Consuls = map[string]*config.ConsulConfig{"default": c.Consul}
+	} else {
+		c.Consul = config.DefaultConsulConfig().Merge(c.Consul)
+		c.Consuls = map[string]*config.ConsulConfig{"default": c.Consul}
```

Comment:
These tests were really confusing to me, as they mix file parsing and setting default values.

I had to add this logic because `basicConfig` already has a `consul` block, so the Consul's default were not being set. 

Since `TemplateIdentity.Name` is one of these defaults, and cannot be set in the config file, `TestConfig_Parse/basic.hcl` and `TestConfig_Parse/basic.json` would fail because `basicConfig` doesn't have the identity name set, while `actual = oldDefault.Merge(actual)` will set the proper name.

With this change, the call to `basicConfig.addDefaults()` will set the service name.

(side note: it's kind of scary that we're modifying a global variable inside a test case 😬)